### PR TITLE
Repro #28106 - Scrolling up on schema list jumps to the top when creating a new question

### LINKED ISSUE #28106: Scrolling up on schema list jumps to the top when creating a new question
**Describe the bug**
When creating a new question, I selected Raw Data and then clicked on a database (Postgres) that shows a list of schema -- it has at least 30 schemas. When scrolling down to view the list of schema, it seems fine, but as soon as you try scrolling up, it jumps to the top of the list instead of scrolling smoothly. 

**Logs**
n/a

**To Reproduce**
1. Create a database with at least 30 schemas. A small # like 2 won't surface the issue. 
2. Create a new question and browse to the database to show the list of schemas
3. Scroll down to view the end of the schema list
4. Scroll up and it will jump to the top of the list instantly. 

**Expected behavior**
When scrolling up, it should scroll up smoothly just like scrolling down. 

**Screenshots**
n/a

**Information about your Metabase Installation:**

You can get this information by going to Admin -> Troubleshooting.

- Your browser and the version: Chrome
- Your operating system: OS X
- Your databases: Postgres
- Metabase version: master
- Metabase hosting environment: Metabase Cloud Dev
- Metabase internal database: Postgres

**Severity**
How severe an issue is this bug to you? Is this annoying, blocking some users, blocking an upgrade or blocking your usage of Metabase entirely?
It's a bit annoying and also makes it harder to find what you are looking for if you didn't find it when scrolling down. 

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by kamilmielnik**
Closes #28106

Initially I thought that we should just close the issue (because we no longer use the old component that had the bug) but working on the repro helped [uncover a different scrolling issue](https://github.com/metabase/metabase/issues/44966) in the new Data Picker component, so ...

Stress test: https://github.com/metabase/metabase/actions/runs/9743651704

### CODE REVIEW

File: e2e/test/scenarios/question/notebook-data-source.cy.spec.ts [hunk 1]
```diff
@@ -393,6 +393,57 @@ describe("scenarios > notebook > data source", { tags: "@OSS" }, () => {
...
+      startNewQuestion();
+      entityPickerModal().within(() => {
+        entityPickerModalTab("Tables").click();
+        cy.findByText("Writable Postgres12").click();
+
+        entityPickerModalLevel(1)
+          .findByTestId("scroll-container")
+          .as("schemasList");
+
+        // the list is virtualized and the scrollbar height changes during scrolling (metabase#44966)
```
Comment (kamilmielnik) [inline e2e/test/scenarios/question/notebook-data-source.cy.spec.ts:420]: See #44966

File: e2e/support/test_tables.js [hunk 2]
```diff
@@ -202,3 +202,36 @@ export const multi_schema = async dbClient => {
 
   return schemas;
 };
+
+export const many_schemas = async dbClient => {
```
Comment (kamilmielnik) [inline e2e/support/test_tables.js:206]: Our `multi_schemas` database only has 2 schemas, and this issue can only be reproduced with way more of them.

File: e2e/support/test_tables.js [hunk 3]
```diff
@@ -202,3 +202,36 @@ export const multi_schema = async dbClient => {
...
+  const alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ".split("");
```
Comment (nemanjaglumac) [inline e2e/support/test_tables.js:207]: https://i.imgflip.com/2/2vddr6.jpg

File: e2e/test/scenarios/question/notebook-data-source.cy.spec.ts [hunk 4]
```diff
@@ -393,6 +393,57 @@ describe("scenarios > notebook > data source", { tags: "@OSS" }, () => {
...
+        // simulate scrolling up using mouse wheel 3 times
+        cy.get("@schemasList").realMouseWheel({ deltaY: -100 });
+        cy.wait(100);
+        cy.get("@schemasList").realMouseWheel({ deltaY: -100 });
+        cy.wait(100);
+        cy.get("@schemasList").realMouseWheel({ deltaY: -100 });
+        cy.wait(100);
+
+        // assert first item does not exist - this means the list has not been scrolled to the top
+        cy.findByText("Domestic").should("not.exist");
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/question/notebook-data-source.cy.spec.ts:438]: Is this actually removed from the DOM or only not visible?

Comment (kamilmielnik) [reply e2e/test/scenarios/question/notebook-data-source.cy.spec.ts:438]: Removed because because the list is virtualized.

**Review by nemanjaglumac — APPROVED** [review]
I doubt this will be possible to stress-test, but if it works by any chance, maybe give it a few spins before merging.

**Comment by kamilmielnik** [comment]
> I doubt this will be possible to stress-test, but if it works by any chance, maybe give it a few spins before merging.

There's 404 coming from `api/testing/restore/postgres-writable` so I guess it's not possible ([image](https://github.com/metabase/metabase/assets/6830683/35b4e5a9-b74a-4150-a28e-f3a83a03b289)).

**Pull request merged**